File: program/print-reverse-of-a-number-by-recursion/PrintReverseOfANumberByRecursion.java
import java.util.Scanner;
class RecursionReverse
{
   public static void reverseMethod(int number) {
       if (number < 10) {
	   System.out.println(number);
	   return;
       }
       else {
           System.out.print(number % 10);
           reverseMethod(number/10);
       }
   }
   public static void main(String args[])
   {
	int num=0;
	System.out.print("Please enter a number : ");
	Scanner in = new Scanner(System.in);
	num = in.nextInt();
	System.out.print("Reverse of the given number is: ");
	reverseMethod(num);
	System.out.println();
   }
}

File: program/print-reverse-of-a-number-by-recursion/PrintReverseOfANumberByRecursion.kt
import java.util.Scanner

fun reverseNumber(num: Int): Int {
	return if (num < 10) {
        num
    } else {
        val lastDigit = num % 10
        val remainingNum = num / 10
        val reversedRemainingNum = reverseNumber(remainingNum)
        (lastDigit * Math.pow(10.0, (remainingNum.toString().length).toDouble()).toInt()) + reversedRemainingNum
    }
}

fun main() {
    val reader = Scanner(System.`in`)
    print("Enter a number: ")

    var number:Int = reader.nextInt()

    val reversedNumber = reverseNumber(number)
    println("The reverse of $number is $reversedNumber")
}